File: sdks/php/sdk/src/Types/InputMessage.php
<?php

namespace Zerogpu\Types;

use Zerogpu\Core\Json\JsonSerializableType;
use Zerogpu\Core\Json\JsonProperty;

class InputMessage extends JsonSerializableType
{
    /**
     * @var value-of<InputMessageRole> $role Message author role
     */
    #[JsonProperty('role')]
    public string $role;

    /**
     * @var string $content Message text
     */
    #[JsonProperty('content')]
    public string $content;

    /**
     * @param array{
     *   role: value-of<InputMessageRole>,
     *   content: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->role = $values['role'];
        $this->content = $values['content'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/src/ZerogpuApi/Types/InputMessage.cs
using global::System.Text.Json;
using global::System.Text.Json.Serialization;
using ZerogpuApi.Core;

namespace ZerogpuApi;

[Serializable]
public record InputMessage : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// Message author role
    /// </summary>
    [JsonPropertyName("role")]
    public required InputMessageRole Role { get; set; }

    /// <summary>
    /// Message text
    /// </summary>
    [JsonPropertyName("content")]
    public required string Content { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
